src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -388,38 +417,66 @@ public static boolean isPathIdentifier(String name) {
     return name.contains(MODULE_SLASH);
   }
 
+  /** Removes the leading slash from absolute identifiers */
+  public static String removeAbsoluteIdentifierIndicator(String name) {
+    if (isAbsoluteIdentifier(name)) {
+      return name.substring(1);
+    }
+    return name;
+  }
+
+  /** Converts paths which don't start with either a '.' or '/' to absolute paths */
+  public static String toAbsoluteIdentifier(String name) {
+    if (name.length() == 0 || isAbsoluteIdentifier(name) || isRelativeIdentifier(name)) {
+      return name;
```

Comment:
FYI that path means two completely different things depending on the resolution mode. This has been a source of great confusion to all parties.

```js
import {foo} from 'path/to/bar'
```

In `LEGACY` mode, this path is absolute.
In `NODE` mode, this path is looked up from the registry.
In `BROWSER` mode, this path is invalid.